### PR TITLE
Add api handler for donor request link.

### PR DESCRIPTION
**Pull request by jtuckerk**
remove connect calls to sqlAlchemy connector

### CODE REVIEW

File: client/src/components/Register.js [hunk 1]
```diff
@@ -79,15 +79,28 @@ function Register() {
...
         </div>
       </Col>
     </Container>
   );
 }
 
+function SendDonorRequestLink() {
+  const email = document.getElementById('act-blue-email').value;
+  if ( !email.includes("@") ) {
+    alert("bruh, emails gotta have an @ sign");
```
Comment (trippersham) [inline client/src/components/Register.js:95]: I lol'd, but I'm wondering if we want to hit actual end users with "bruh". Maybe "hey there human, emails..." or something like that?

**Review by trippersham — COMMENTED** [review]
@jtuckerk any chance I can get you to write a few tests for the new DataSink object and request_personal_donation_link route?

File: api/models.py [hunk 2]
```diff
@@ -60,3 +62,7 @@ class Initiative(BaseModel):
     roles: List[VolunteerRole] = []
     events: List[VolunteerEvent] = []
     highlightedItems: List[Union[VolunteerRole,VolunteerEvent]] = []
+
+class PersonalDonationLinkRequest(BaseModel):
+    email: str
```
Comment (trippersham) [inline api/models.py:67]: Looks like Pydantic has [a native email validator](https://pydantic-docs.helpmanual.io/usage/types/#pydantic-types) that we could use. Introduces new error response flows, but ensures data integrity.

File: client/src/components/Register.js (hunk 1 above)
Comment (trippersham) [reply client/src/components/Register.js:95]: Mirroring the comment about the back-end, maybe we do some deeper email validation on the front-end for data integrity. Maybe [this example](https://ui.dev/validate-email-address-javascript/) (whose SEO game is dialed in)?

**Review by ecaoile — COMMENTED** [review]
I'm going to defer on this PR. I'm getting errors again trying to set up the API. This is a bit frustrating...

**Comment by jtuckerk** [comment]
> @jtuckerk any chance I can get you to write a few tests for the new DataSink object and request_personal_donation_link route?

added. these would have been helpful before my last minute lint fixes which broke everything :)

File: api/data_sources.py [hunk 3]
```diff
@@ -97,15 +118,19 @@ def __init__(self, data_source: DataSource, dataset_key: str, primary_key: str,
...
+            except Exception as e:
+                logging.warning(f'failed to get all rows for {self.dataset_key}:\n{e}')
             return results
         return None
     
     def get_row_for_primary_key(self, key: str) -> Optional[RowProxy]:
         if self.data_source.sql_engine:
             query = select([self.sql_table]).where(self.sql_table.c[self.primary_key] == key) # type: ignore
-            results = self.data_source.sql_conn.execute(query).fetchone() # type: ignore
+            results = self.data_source.sql_engine.execute(query).fetchone() # type: ignore
```
[the remark this answers is not in the document]
Comment (jtuckerk) [reply api/data_sources.py:128]: hope the button works

File: client/src/components/Register.js (hunk 1 above)
Comment (jtuckerk) [reply client/src/components/Register.js:95]: sooo boring. i updated it.

**Pull request merged**